[DO NOT MERGE] Avoid install & uninstall of setup package if local_ci_setup set

tom--pollard opened the pull request:
Also remove pip from `conda install` list in build templates
as already a transient dependency from conda-build.

This should remove some redundant time in certain pipeline configurations.

beckermr commented on news/avoid_setup_package_uninstall:
```diff
@@ -0,0 +1,24 @@
+
+**Removed:**
+
+* Removed pip from `conda install` list in build templates as already a transient dependency from conda-build
```
We should never rely on transient deps. If one day `conda-build` does not depend on pip, then we will get an error. Any thing we explicitly use should be listed.

tom--pollard commented:
@beckermr the uninstall also removes the dependencies on the setup package? Unless I'm missing something

beckermr commented:
It does not appear to do so: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=194893&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=186

tom--pollard commented:
Ok, so it looks like the `--force` is cause this behaviour:
```
(deleteme) tom@x260:/home$ conda uninstall --force conda-forge-ci-setup

## Package Plan ##

  environment location: /home/tom/conda-project/miniforge3/envs/deleteme

  removed specs:
    - conda-forge-ci-setup


The following packages will be REMOVED:

  conda-forge-ci-setup-3.2.0-py38h3611fdf_0


Proceed ([y]/n)? n


CondaSystemExit: Exiting.

(deleteme) tom@x260:/home$ conda uninstall conda-forge-ci-setup
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/tom/conda-project/miniforge3/envs/deleteme

  removed specs:
    - conda-forge-ci-setup


The following packages will be REMOVED:

  anaconda-client-1.7.2-py_0
  attrs-19.3.0-py_0
  c-ares-1.16.1-h516909a_0
  click-7.1.2-pyh9f0ad1d_0
  clyent-1.2.2-py_1
  conda-env-2.6.0-1
  conda-forge-ci-setup-3.2.0-py38h3611fdf_0
  curl-7.71.1-he644dc0_4
  decorator-4.4.2-py_0
  expat-2.2.9-he1b5a44_2
  gettext-0.19.8.1-hc5be6a0_1002
  git-2.28.0-pl526h5e3e691_1
  importlib-metadata-1.7.0-py38h32f6830_0
  importlib_metadata-1.7.0-0
  ipython_genutils-0.2.0-py_1
  jq-1.6-h14c3975_1000
  jsonschema-3.2.0-py38h32f6830_1
  jupyter_core-4.6.3-py38h32f6830_1
  krb5-1.17.1-hfafb76e_2
  libcurl-7.71.1-hcdd3856_4
  libedit-3.1.20191231-h46ee950_1
  libev-4.33-h516909a_0
  libnghttp2-1.41.0-hab1572f_1
  libssh2-1.9.0-hab1572f_5
  nbformat-5.0.7-py_0
  oniguruma-6.9.3-h516909a_0
  pcre-8.44-he1b5a44_0
  perl-5.26.2-h516909a_1006
  pyrsistent-0.16.0-py38h1e0a361_0
  python-dateutil-2.8.1-py_0
  shyaml-0.6.1-py_0
  traitlets-4.3.3-py38h32f6830_1
  zipp-3.1.0-py_0


Proceed ([y]/n)? 
```
I would not expect this behaviour given the help text of `--force`
```
  --force-remove, --force
                        Forces removal of a package without removing packages
                        that depend on it. Using this option will usually
                        leave your environment in a broken and inconsistent
                        state.
```

I'm not sure in what world curl depends on it :)

beckermr commented:
We good to close this @tom--pollard?

tom--pollard commented:
@beckermr Sure, I'll look to work around it in a separate PR for https://github.com/conda-forge/conda-smithy/issues/1366

tom--pollard commented:
I've raised https://github.com/conda/conda/issues/10148 on conda for the strange uninstall behaviour
